core_api_c_binding/src/lib.rs

```diff
@@ -50,29 +49,35 @@ pub unsafe extern "C" fn holochain_new(ptr: *mut Dna) -> *mut Holochain {
     }
 }
 
-fn get_context() -> Result<Context, HolochainError> {
-    let agent = Agent::from("c_bob".to_string());
-    match UserDirs::new() {
-        Some(user_dir) => {
-            let home_dir = user_dir.home_dir();
-            let cas_path = storage_path(home_dir, "cas")?;
-            let eav_path = storage_path(home_dir, "eav")?;
-            create_path_if_not_exists(&cas_path)?;
-            create_path_if_not_exists(&eav_path)?;
-            Context::new(
-                agent,
-                Arc::new(Mutex::new(NullLogger {})),
-                Arc::new(Mutex::new(SimplePersister::new())),
-                FilesystemStorage::new(&cas_path).unwrap(),
-                EavFileStorage::new(eav_path).unwrap(),
-            )
-        }
-        None => Err(HolochainError::ErrorGeneric(
-            "Could not create context".to_string(),
-        )),
+#[no_mangle]
+pub unsafe extern "C" fn holochain_load(storage_path: CStrPtr) -> *mut Holochain {
+    let path = CStr::from_ptr(storage_path).to_string_lossy().into_owned();
+    let context = get_context(&path);
+
+    match context {
+        Ok(con) => match Holochain::load(path, Arc::new(con)) {
+            Ok(hc) => Box::into_raw(Box::new(hc)),
+            Err(_) => std::ptr::null_mut(),
+        },
+        Err(_) => std::ptr::null_mut(),
     }
 }
 
+fn get_context(path: &String) -> Result<Context, HolochainError> {
+    let agent = Agent::from("c_bob".to_string());
+    let cas_path = format!("{}/cas", path);
+    let eav_path = format!("{}/eav", path);
+    create_path_if_not_exists(&cas_path)?;
+    create_path_if_not_exists(&eav_path)?;
+    Context::new(
+        agent,
+        Arc::new(Mutex::new(NullLogger {})),
+        Arc::new(Mutex::new(SimplePersister::new())),
+        FilesystemStorage::new(&cas_path).unwrap(),
```

Comment:
How does unwrap work when interacting with c extern functions? Can we use question mark instead?

Reply:
I only changed the indentation of this line. The C string is converted in line 54.

Reply:
:+1: